test/column/resizable.test.tsx

```diff
@@ -68,34 +51,85 @@ const columns: readonly Column<never>[] = [
+  await resize({ column: col1, resizeBy: -50 });
   expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px' });
 });
 
-test('should resize column when dragging the handle', () => {
-  setup({ columns, rows: [] });
+test('should resize column when dragging the handle', async () => {
+  setup<Row, unknown>({ columns, rows: [] });
   const [, col2] = getHeaderCells();
-  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px' });
-  resize({ column: col2, clientXStart: 289, clientXEnd: 250, rect: { right: 300, left: 100 } });
-  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 161px' });
+  const grid = getGrid();
+  expect(grid).toHaveStyle({ gridTemplateColumns: '100px 200px' });
+  await resize({ column: col2, resizeBy: -50 });
+  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 150px' });
 });
 
-test('should use the maxWidth if specified', () => {
-  setup({ columns, rows: [] });
+test('should use the maxWidth if specified', async () => {
+  setup<Row, unknown>({ columns, rows: [] });
   const [, col2] = getHeaderCells();
-  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px' });
-  resize({ column: col2, clientXStart: 295, clientXEnd: 1000, rect: { right: 300, left: 100 } });
+  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px ' });
+  await resize({ column: col2, resizeBy: 1000 });
   expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 400px' });
 });
 
-test('should use the minWidth if specified', () => {
-  setup({ columns, rows: [] });
+test('should use the minWidth if specified', async () => {
+  setup<Row, unknown>({ columns, rows: [] });
   const [, col2] = getHeaderCells();
   expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px' });
-  resize({ column: col2, clientXStart: 295, clientXEnd: 100, rect: { right: 300, left: 100 } });
+  await resize({ column: col2, resizeBy: -150 });
   expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 100px' });
 });
+
+test('should not auto resize column if resizable is not specified', async () => {
+  setup<Row, unknown>({
+    columns,
+    rows: [
+      {
+        col1: 1,
+        col2: 'a'.repeat(50)
+      }
+    ]
+  });
+  const [col1] = getHeaderCells();
+  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px' });
+  await autoResize(col1);
+  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px' });
+});
+
+test('should auto resize column when resize handle is double clicked', async () => {
+  setup<Row, unknown>({
+    columns,
+    rows: [
+      {
+        col1: 1,
+        col2: 'a'.repeat(50)
+      }
+    ]
+  });
+  const [, col2] = getHeaderCells();
+  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 200px' });
+  await autoResize(col2);
+  expect(getGrid()).toHaveStyle({ gridTemplateColumns: '100px 327.703px' });
+});
+
+test('should use the maxWidth if specified on auto resize', async () => {
```

Comment:
Could also test minWidth

Reply:
I will add it in the nest PR
https://github.com/adazzle/react-data-grid/pull/3547